connectivity/cellular/tests/UNITTESTS/framework/AT/CMakeLists.txt

```diff
@@ -1,9 +1,13 @@
 # Copyright (c) 2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-add_subdirectory(at_cellularcontext)
-add_subdirectory(at_cellulardevice)
-add_subdirectory(at_cellularinformation)
-add_subdirectory(at_cellularnetwork)
-add_subdirectory(at_cellularsms)
-add_subdirectory(at_cellularstack)
+list(APPEND MBED_ADD_UNITTEST_SUBDIRS
+    at_cellularcontext
+    at_cellulardevice
+    at_cellularinformation
+    at_cellularnetwork
+    at_cellularsms
+    at_cellularstack
+)
+
+mbed_add_all_subdirectories("${MBED_ADD_UNITTEST_SUBDIRS}")
```

Comment:
I recall seeing this in another PR, do we need this and not to use just pure CMake?

Reply:
Yes, we have used this macro in storage PR, This macro added to avoid repetitively call add_subdirectory for example in lorawan unittest we are in a position to add a subdirectory of 17 dir, so better to use this macro

Reply:
lets review the macro once we get the entire picture (feature branch ready)

Reply:
I would rather keep it pure CMake.